src/Core/tests/UnitTests/Layouts/GridLayoutManagerTests.cs

```diff
@@ -3235,5 +3289,401 @@ public void StarsAdjustWhenArrangeAndMeasureWidthDiffer(double widthConstraint,
+			// First and third columns (2*) should be approximately twice the width of the middle column (1*)
+			Assert.True(rect0.Width > rect1.Width * 1.5, $"First column ({rect0.Width}) should be larger than middle column ({rect1.Width})");
+			Assert.True(rect2.Width > rect1.Width * 1.5, $"Third column ({rect2.Width}) should be larger than middle column ({rect1.Width})");
+
+			// First and third columns should be approximately equal
+			Assert.Equal(rect0.Width, rect2.Width, 1);
+
+			// Total should not exceed constraint
+			var totalWidth = rect0.Width + rect1.Width + rect2.Width;
+			Assert.True(totalWidth <= widthConstraint + 1, $"Total width {totalWidth} should not exceed constraint {widthConstraint}");
+
+			// Convert Dp values to pixels for verification
+			var pixelWidth0 = rect0.Width * density;
+			var pixelWidth1 = rect1.Width * density;
+			var pixelWidth2 = rect2.Width * density;
+
+			// With density-aware distribution, pixel values should be exact integers
+			Assert.Equal(Math.Round(pixelWidth0), pixelWidth0);
+			Assert.Equal(Math.Round(pixelWidth1), pixelWidth1);
+			Assert.Equal(Math.Round(pixelWidth2), pixelWidth2);
+		}
+
+		[Fact]
+		[Category(GridStarSizing)]
+		public void StarLayoutPreventsOverflow()
+		{
+			// Test that demonstrates the problem being solved: ensuring no overflow occurs
+			var grid = CreateGridLayout(columns: "*, *, *, *, *"); // 5 equal columns
+
+			var views = new IView[5];
+			for (int i = 0; i < 5; i++)
+			{
+				views[i] = CreateTestView(new Size(20, 50));
+				SetLocation(grid, views[i], col: i);
+			}
+			SubstituteChildren(grid, views);
+
+			// Use a constraint that would cause rounding issues
+			var widthConstraint = 333.33; // Doesn't divide evenly by 5
+			
+			// Set up capture for all view rectangles
+			var rects = new Rect[5];
+			for (int i = 0; i < 5; i++)
+			{
+				int index = i; // Capture loop variable for closure
+				views[i].When(x => x.Arrange(Arg.Any<Rect>())).Do(x => rects[index] = x.Arg<Rect>());
+			}
+			
+			MeasureAndArrangeFixed(grid, widthConstraint, 100);
+
+			// Collect all arranged rects
+
+			// Verify sequential layout
+			for (int i = 1; i < 5; i++)
+			{
+				Assert.True(rects[i].X >= rects[i - 1].X + rects[i - 1].Width - 1, $"Column {i} should start after column {i - 1}");
+			}
+
+			// Most importantly: verify no overflow
+			var totalWidth = rects.Sum(r => r.Width);
+			Assert.True(totalWidth <= widthConstraint + 1, $"Total width {totalWidth} should not exceed constraint {widthConstraint}");
+
+			// Verify all space is used (no significant gaps)
+			Assert.True(totalWidth >= widthConstraint - 5, $"Total width {totalWidth} should be close to constraint {widthConstraint}");
+		}
+
+		[Theory]
+		[InlineData(1)]
+		[InlineData(2)]
+		[InlineData(3)]
+		[InlineData(4)]
+		[InlineData(5)]
+		[InlineData(6)]
+		[InlineData(7)]
+		[InlineData(8)]
+		[InlineData(9)]
+		[InlineData(10)]
+		[InlineData(11)]
```

Comment:
Add the expected outcome of sizes for each child view to each inlinedata. 

So the inline data should look something like

[InlineData(4, [201,201,202,202] )]

This way we can visually see what the expected outcomes are with each distribution